tests/lib/rules/accessor-pairs.js

```diff
@@ -18,26 +18,301 @@ const rule = require("../../../lib/rules/accessor-pairs"),
+            options: [{ setWithoutGet: true, getWithoutSet: true }]
+        },
+        {
+            code: "var o = { get a() {}, b, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { get a() {}, ...b, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 2018 }
+        },
+        {
+            code: "var o = { get a() {}, set a(foo) {}, ...a };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 2018 }
+        },
+
+        // Duplicate keys. This is the responsibility of no-dupe-keys, but this rule still checks is there the other accessor kind.
+        {
+            code: "var o = { get a() {}, get a() {}, set a(foo) {}, };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { get a() {}, set a(foo) {}, get a() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { get a() {}, set a(foo) {}, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { set a(bar) {}, get a() {}, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { get a() {}, get a() {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: false }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { set a(foo) {}, set a(foo) {} };",
+            options: [{ setWithoutGet: false, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { get a() {}, set a(foo) {}, a };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var o = { a, get a() {}, set a(foo) {} };",
+            options: [{ setWithoutGet: true, getWithoutSet: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+
+        /*
+         * This should be actually invalid by this rule!
+         * This code creates a property with the setter only, the getter will be ignored.
+         * It's treated as 3 attempts to define the same key, and the last wins.
+         * However, this edge case is not covered, it should be reported by no-dupe-keys anyway.
+         */
+        {
+            code: "var o = { get a() {}, a:1, set a(foo) {} };",
```

Comment:
Do we want to add this to the "known limitations "documentation as well? I could see someone reporting this as a bug in the future.

Reply:
I don't want to block this from landing in this release, so I'm going to go ahead and merge this. If we decide the documentation could be improved, we can always do that in a follow up PR.

Reply:
Was about to reply when you merged :)

I think this is a good idea, I'll make a small docs PR to add this.

Reply:
Sounds good!